spec/models/relationship_spec.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Relationship, type: :model do
+  describe 'validations' do
+    let(:user1) { create(:user) }
+    let(:user2) { create(:user) }
+    let(:relationship) { build(:relationship, follower: user1, followed: user2) }
+
+    it 'should be valid' do
+      expect(relationship).to be_valid
+    end
+
+    it 'should require a follower_id' do
+      relationship.follower_id = nil
+      expect(relationship).not_to be_valid
+    end
+
+    it 'should requiere a followed_id' do
+      relationship.followed_id = nil
+      expect(relationship).not_to be_valid
+    end
+
+    it 'should follow a user' do
+      expect(user2.following?(user1)).to be_falsey
+      user2.follow(user1)
+      expect(user2.following?(user1)).to be_truthy
+      expect(user1.followers.include?(user2)).to be_truthy
+    end
+
+    it 'should unfollow a user' do
+      user1.follow(user2)
+      user1.unfollow(user2)
+      expect(user1.following?(user2)).to be_falsey
+    end
+  end
```

Comment:
The structure of this test is a bit off. The describe block defined is for validations but there are tests for model methods.

Apart from that, please do the validations with shoulda matchers, as you did with other models.